fmriprep/workflows/bold/base.py

```diff
@@ -141,52 +143,59 @@ def init_func_preproc_wf(bold_file):
     from niworkflows.interfaces.utils import DictMerge
     from sdcflows.workflows.base import init_sdc_estimate_wf, fieldmap_wrangler
 
-    ref_file = bold_file
     mem_gb = {'filesize': 1, 'resampled': 1, 'largemem': 1}
     bold_tlen = 10
-    multiecho = isinstance(bold_file, list)
 
     # Have some options handy
-    layout = config.execution.layout
     omp_nthreads = config.nipype.omp_nthreads
     freesurfer = config.workflow.run_reconall
     spaces = config.workflow.spaces
     output_dir = str(config.execution.output_dir)
 
+    # Extract BIDS entities and metadata from bold reference file
+    ref_file = select_first(bold_file)
+    layout = config.execution.layout
+    entities = layout.parse_file_entities(ref_file)
+    metadata = layout.get_metadata(ref_file)
+
+    # Drop echo entity for future queries, have a boolean shorthand
+    multiecho = bool(entities.pop("echo", None))
+    if multiecho and (isinstance(bold_file, str) or len(bold_file) == 1):
+        multiecho = False
+        config.loggers.warning(
+            f"Only one echo <{ref_file}> found in a seemingly multi-echo dataset. "
+            "Falling back to single-echo mode."
+        )
```

Comment:
This would definitely be the case if `echo-idx` was passed, right? Should it raise a warning in that case?

Reply:
You mean ``--echo <idx>``? Yeah, although it seems like a tricky case - do we really want to allow people to process just one echo?

I am happy to account for that, but let's first think about whether selecting echoes should be allowed.

Reply:
The argument is `--echo-idx <idx>`.

If you want to analyze multi-echo like single-echo, it makes sense to just preprocess the echo time closest to a typical single-echo echo time (e.g., 30ms for 3T). That said, I'm not sure how common that is.

Reply:
Well, it's a use-case. Probably not very common but reasonable. Okay, I'll change the message for that situation.

There is another anomalous use case: two echoes. How should fMRIPrep proceed in that case?

Reply:
@emdupre can correct me, but I think the best move would be to process both echoes in parallel without trying to combine them. That way, folks could apply the Bright & Murphy denoising method if they want to, after preprocessing. In that method, you just regress the shorter echo out of the longer echo, so it shouldn't be that big of a deal to wait until after preprocessing to do it.

Reply:
Okay, that will require some refinement here: https://github.com/nipreps/niworkflows/blob/master/niworkflows/utils/bids.py#L199-L201 so that two echoes are returned separately as if they were just BOLD runs.

Reply:
If you all agree, let's address this on another PR and get this one finished.